rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,450 @@
+
+## Adoption strategy
+
+1. Storybook guideline doc for understanding how to work with layouts in Blade, talking about best practices and common examples.
+1. This is a progressive, non-breaking change (we can release components incrementally) which teams should be able to adopt on their own since it should drastically simplify building layouts.
+1. Doesn't need separate prioritization from product, engineers can start using this for newer layouts and gradually replace their existing layout solutions (eg. their custom `Box` implementation). During this migration phase we might receive feedback on some edge cases that might not be smoothly addressed by layout primitives.
+
+## How do we educate people?
+
+- Add a guideline doc in storybook which talks about layout strategy with some sample examples (covering all supported APIs with best practices), include this rfc for background reading.
+- TS docs and auto suggestions so people don't need to always refer documentation site. These should be correctly reflected for React and React Native.
+
+## Open Questions
+
+**Q1.** Should the base component be called `Box` or something else like `Layout`, `View`, etc.
+
+| Box                                                                       | Layout                                                      | View                                                |
+| ------------------------------------------------------------------------- | ----------------------------------------------------------- | --------------------------------------------------- |
+| Well known terminology amongst existing systems                           | New terminology                                             | Seen across some implementations                    |
+| Sounds quite generic                                                      | Sounds layout specific                                      | Sounds native only - related to React Native `View` |
+| Users might expect this to support all CSS properties as in other systems | Can be reasoned to be limited to layout specific properties | Similar to `Box`                                    |
+
+> There are only two hard problems in software engineering - cache invalidation, naming things and off by one error.
+
+Considering the scope of `Box` may be expected to increase over time I'm leaning towards going ahead with `Box`. This also has an added benefit of incremental migration _(new properties will be available on this component)_. Plus it's a relatively well known terminology in the community. `Layout` does address the issue of reasoning about restricted set of properties but if we come across the need for having other CSS properties _(as usecases grows)_ they won't fit within `Layout` and we may need other components to support them. So, just by calling this primitive `Layout` it won't necessarily contain the need for other properties in building UIs.
+
```

Comment:
Should we add a short answer here based on the discussion?

Reply:
Done in https://github.com/razorpay/blade/pull/950/commits/a76d5aa03125cf9229ca3faa754711666374cc5f